### PR TITLE
Feature 190 did individual page

### PR DESCRIPTION
**Pull request by ThePhisch**
Added `DID Individual Page`

This PR introduces the `PageDID` component, which is the page dedicated to serving information on one individual DID. The intent is that users can access this page via a permalink, following the link on `ListDID`, or by having an exact match in the search function.

The `PageDID` component has several subpages, the composition of subpages depends on the DID Type of the DID. They are:

- Dataset
    - *Rules*: Show rule with state (`RuleStateTag`), the associated account, subscription and last modification date.
    - *Dataset Replicas*: Shows the RSEs to which the dataset is replicated and the availability (and available files, available bytes, creation date, last access date). The availability is shown as a tiny `FileReplicaState[unavailable]` tag, which takes the shape of a circle. Uses the `RSETag` to show whether an RSE has been blocklisted for any combinatio of read/write/delete.
    - *File Replica States [Dataset Version]*: Two tables, set to stack horizontally not vertically on the widest screens. Select a (file) DID on the first table, then view the file replica states of this file on the second table. Uses the `FileReplicaState` tag.
    - *Metadata*: Metadata specific to this DID that is not generally set for all DIDs (custom key-value pairs), shown in a two-column table. Specific values are formatted into tags, such as dates `DateTag`, booleans `BoolTag`, null `NullTag`, etc.
- File
    - *File Replica States [File Version]*: similar to the dataset version, but without the first table to select a file DID.
    - *Parent DIDs:* shows a list of DIDs which identify as the parents of this file, including their DID Type. ********note******** that the DIDType might be superfluous information, since containers must not directly contain files.
    - *Metadata*
- Container
    - *Contents*: shows a list of DIDs which identify as children of this container, including their DID type.
    - *Rules*
    - *File Replica States [Container Version]* **********note********** that this tab entry should probably be done away with
    - *Metadata*

The tables are filterable (by string or by state) and sortable where it is logical to add these controls. It is not possible to filter/sort for RSE blocklist state. It is *not yet* possible to filter the subscriptions.

A number of new tags were created in the process of creating this PR. Their designs are simple and intend to stay within the pastel colour range. The selection of colours is taken from the standard colour palette tailwind provides (no customs provided). They are:
- CheckmarkTag (unused)
- DateTag
- NullTag
- RSEBlockTag (components of `RSETag`)
- RSETag (showing the RSE name and adding the `RSEBlockTag`s)
- ReplicaStateTag (_Available_, _Unavailable_, _Copying_, ...)
- RuleStateTag (_Replicating_, _OK_, _Stuck_, ...)

In many cases, new states have been added to `core/entity/rucio.ts` reflecting enums from the rucio database core.

Besides the subpages (which can be selected via a `Tabs` component), the page consists of a block lying above the subpages. This block will give the title of the page and a quick summary of the DID, similar to the `ListDID` right-hand-side pane (it is the same component, `DIDMetaView`, but this time the component is allowed to stack horizontally, in order to not use as much vertical space on wide screens).

For the case that this page (individual DID page) has been accessed via the `ListDID`-component summarising the results of a previous search, a `Back to DID List` button will be shown, taking the user back to the list page. The idea is that the search results will remain cached and do not have to be queried again.

## Followups

- *Unclear design language in linking* to other DID/RSE/Rule-pages. So far, several different ways of linking these individual pages have been tried. Most notably the `link` icon to the right of the name string. I believe it is more logical to move away from this and simply use the string as a link (and also decorate it with an underline when hovering over it) in the future, as can be seen in the `Dataset Replicas` subpage for datasets. Note that this would mean that it is no longer possible to select the entire string with one click (i.e. using the `select-all` tailwind class) for people who want to select the string, copy it and then paste it into the searchbar.
- *Changing the table* definitions to the *new architecture* of having cell and header definitions within the `columnDef` array (instead of having the cell definitions in the array and manually rendering the headers). This would allow for more sophisticated conditional column hiding/showing using the tanstack API. Need to find a better and less "hacky" way to deal with the `FullFilter` string filtering though (at the moment, the `column.id` must be the string shown on top of the column as it is rendered to the reader).
- **************************dealing with subscriptions************************** in the `Rule` tables needs to be handled differently. It would make sense to link to a specific `Subscription` page and make the columns string-searchable. This is especially important since we know that each rule must only be associated to one subscription. Perhaps we should create a tag.
- allow the summary of DID information to be an accordion that can be folded in and out depending on the user’s needs.
- Issues in responsive design, including but not limited to
    - The summary of DID information displayed at the top of the page looks weird for files
    - On certain medium-width screens, the tab elements such as `Dataset Replicas` and `File Replica States` will take up 2 lines, meaning that their height is greater than the surrounding tab elements, which looks plain ugly
    - While all tables should work on even the smallest widths, this was sometimes achieved by omitting columns without adding the option of selecting a row and showing the missing information in an infobox below (as would be the case in `Dataset Replicas`).
    - Tables on mobile should have their height adjusted so that they can be read without scrolling. Having both pagination and scrolling is bad design.
- Should the individual DID page be accessed from a different page, it should also be possible to return to that page via the `Back to DID List` button, it will be necessary to change the string in that button though.
- Complete reorganising common components for streamed tables into `components/StreamedTables`
- Migrate custom types away from `rucio-dto` and other false locations and collect them in `core/entity/rucio.ts` and other sensible places.

### CODE REVIEW

File: src/lib/core/entity/rucio.ts [hunk 1]
```diff
@@ -59,3 +59,14 @@ export type RSE = {
     rse_type: string,
     volatile: boolean,
 }
+
+// rucio.db.sqla.constants::ReplicaState
+export type ReplicaState = "Available" | "Unavailable" | "Copying" | "Being_Deleted" | "Bad" | "Temporary_Unavailable"
```
Comment (maany) [inline src/lib/core/entity/rucio.ts:64]: Can you change these to be an `enum`?

File: src/lib/infrastructure/data/view-model/pagedid.d.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { ReplicaState, RSEBlockState } from "@/lib/core/entity/rucio"
+import { RuleState } from "@/lib/core/entity/rucio"
+import { ObjectFlags } from "typescript"
+
+export type FilereplicaStateD = {
+    scope: string,
+    name: string,
+    Available: number,
```
Comment (maany) [inline src/lib/infrastructure/data/view-model/pagedid.d.ts:8]: should be lower case? for consistency

File: src/lib/infrastructure/data/view-model/createRule.d.ts
[the remark this answers is not in the document]
Comment (maany) [reply src/lib/infrastructure/data/view-model/createRule.d.ts:1]: at least for the api layer. :) Also, have you used any particular naming conventions for the components?

File: tools/create-component-story
Comment (maany) [inline tools/create-component-story:1]: Noice!

File: src/lib/infrastructure/data/view-model/pagedid.d.ts (hunk 2 above)
Comment (maany) [reply src/lib/infrastructure/data/view-model/pagedid.d.ts:8]: Could you document that `D` stands for here?

File: src/lib/infrastructure/data/view-model/createRule.d.ts
[the remark this answers is not in the document]
Comment (ThePhisch) [reply src/lib/infrastructure/data/view-model/createRule.d.ts:1]: Components need to begin with a capital letter for react to accept them. I have stuck to kebab-case

**Pull request merged**